File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/aberturacaixa/AberturaCaixaService.java
package br.com.grupopipa.gestaointegrada.financeiro.aberturacaixa;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.atendimento.lancamento.dto.LancamentoFinanceiroGridDTO;
import br.com.grupopipa.gestaointegrada.financeiro.movimentacaocaixa.MovimentacaoCaixaGridDTO;

public interface AberturaCaixaService {

    AberturaCaixaDTO abrir(AbrirCaixaRequest request);

    AberturaCaixaDTO fechar(UUID id, FecharCaixaRequest request);

    Optional<AberturaCaixaDTO> findAtivaByCaixaId(UUID caixaId);

    List<CaixaComStatusDTO> listarMeusCaixas();

    CaixaComStatusDTO statusPorCaixa(UUID caixaId);

    List<LancamentoFinanceiroGridDTO> listarLancamentosPendentes(UUID aberturaCaixaId);

    List<MovimentacaoCaixaGridDTO> listarMovimentacoes(UUID aberturaCaixaId);
}

File: src/frontend/src/app/components/financeiro/operacao-caixa/abertura-caixa.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AberturaCaixaDTO } from './model/abertura-caixa-dto';
import { CaixaComStatusDTO } from './model/caixa-com-status-dto';
import { MovimentacaoCaixaDTO } from './model/movimentacao-caixa-dto';
import { LancamentoFinanceiroGridDTO } from '../../atendimento/lancamento/model/lancamento-financeiro-grid-dto';

@Injectable()
export class AberturaCaixaService {
  private readonly baseUrl = '/api/abertura-caixa';
  private http = inject(HttpClient);

  abrir(caixaId: string, valorAbertura: number): Observable<AberturaCaixaDTO> {
    return this.http.post<AberturaCaixaDTO>(`${this.baseUrl}/abrir`, {
      caixaId,
      valorAbertura,
    });
  }

  fechar(id: string, valorConferencia: number, observacoes: string): Observable<AberturaCaixaDTO> {
    return this.http.post<AberturaCaixaDTO>(`${this.baseUrl}/${id}/fechar`, {
      valorConferencia,
      observacoes,
    });
  }

  listarMeusCaixas(): Observable<CaixaComStatusDTO[]> {
    return this.http.get<CaixaComStatusDTO[]>(`${this.baseUrl}/meus-caixas`);
  }

  statusPorCaixa(caixaId: string): Observable<CaixaComStatusDTO> {
    return this.http.get<CaixaComStatusDTO>(`${this.baseUrl}/status/${caixaId}`);
  }

  listarLancamentosPendentes(aberturaCaixaId: string): Observable<LancamentoFinanceiroGridDTO[]> {
    return this.http.get<LancamentoFinanceiroGridDTO[]>(
      `${this.baseUrl}/${aberturaCaixaId}/lancamentos-pendentes`
    );
  }

  listarMovimentacoes(aberturaCaixaId: string): Observable<MovimentacaoCaixaDTO[]> {
    return this.http.get<MovimentacaoCaixaDTO[]>(
      `${this.baseUrl}/${aberturaCaixaId}/movimentacoes`
    );
  }
}
